wormhole-connect/src/views/v2/Bridge/AmountInput/index.tsx

```diff
@@ -126,7 +126,7 @@ const AmountInput = (props: Props) => {
     }
 
     setValidationResult(amountValidation);
-  }, [tokenAmount, validationResult]);
+  }, [tokenAmount, tokenBalance, route, validationResult]);
```

Comment:
These were missing in validation checks, hence not checking amount when route and token changes (which causes a change in tokenBalance value).